nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -205,177 +207,186 @@ public void onScheduled(ProcessContext context) throws Exception {
+                                return null;
+                            default:
+                                break;
                         }
-                        createdFile = tempCopyFile;
-                        BufferedInputStream bis = new BufferedInputStream(in);
-                        StreamUtils.copy(bis, fos);
-                        bis = null;
-                        fos.flush();
-                    } finally {
-                        try {
-                            if (fos != null) {
-                                fos.close();
-                            }
-                        } catch (RemoteException re) {
-                            // when talking to remote HDFS clusters, we don't notice problems until fos.close()
-                            if (createdFile != null) {
+                    }
+
+                    // Write FlowFile to temp file on HDFS
+                    final StopWatch stopWatch = new StopWatch(true);
+                    session.read(putFlowFile, new InputStreamCallback() {
+
+                        @Override
+                        public void process(InputStream in) throws IOException {
+                            OutputStream fos = null;
+                            Path createdFile = null;
+                            try {
+                                if (conflictResponse.equals(APPEND_RESOLUTION_AV.getValue()) && destinationExists) {
+                                    fos = hdfs.append(copyFile, bufferSize);
+                                } else {
+                                    fos = hdfs.create(tempCopyFile, true, bufferSize, replication, blockSize);
+                                }
+                                if (codec != null) {
+                                    fos = codec.createOutputStream(fos);
+                                }
+                                createdFile = tempCopyFile;
+                                BufferedInputStream bis = new BufferedInputStream(in);
+                                StreamUtils.copy(bis, fos);
+                                bis = null;
+                                fos.flush();
+                            } finally {
                                 try {
-                                    hdfs.delete(createdFile, false);
+                                    if (fos != null) {
+                                        fos.close();
+                                    }
+                                } catch (RemoteException re) {
+                                    // when talking to remote HDFS clusters, we don't notice problems until fos.close()
+                                    if (createdFile != null) {
+                                        try {
+                                            hdfs.delete(createdFile, false);
+                                        } catch (Throwable ignore) {
+                                        }
```

Comment:
I can add a logging statement, this wasn't really changed though, it was existing code that I simply wrapped in doAs(...)